email/pardot.go

```diff
@@ -56,36 +67,42 @@ type oAuthToken struct {
 	expiresAt   time.Time
 }
 
-// PardotClientImpl handles authentication and sending contacts to Pardot. It
-// implements the PardotClient interface.
-type PardotClientImpl struct {
+// SalesforceClientImpl handles authentication and sending contacts to Pardot
+// and creating Cases in Salesforce.
+type SalesforceClientImpl struct {
 	businessUnit string
 	clientId     string
 	clientSecret string
-	endpointURL  string
+	pardotURL    string
+	casesURL     string
 	tokenURL     string
 	token        *oAuthToken
 	clk          clock.Clock
 }
 
-var _ PardotClient = &PardotClientImpl{}
+var _ SalesforceClient = &SalesforceClientImpl{}
```

Comment:
Is there a reason you're using this shape of interface assertion? I think the below is more idiomatic, and zero-allocation:
```suggestion
var _ SalesforceClient = (*SalesforceClientImpl)(nil)
```